### PR TITLE
Made the Error message userfriendly

### LINKED ISSUE #6269: [mesheryctl] improve the error messages shown to users for "mesheryctl perf"
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._ For additional context, see https://github.com/meshery/meshery/pull/5661.

#### Current Behavior
Responses from `mesheryctl` commands don't always have the best user experience in mind. 

#### Desired Behavior
Improve the error messages in mesheryctl commands from the user's perspective. 
1. Ensure that there are proper error messages when users:
    1. Give more arguments to a command than required
    1. Misspell a command
    1. Do not give the required input via flags or args
    1. Use the command without any arguments or flags.
1. The error messages being displayed to users are meaningful from user's perspective. And if applicable, a potential fix for the problem should be shown too.  For example
    1. "failed to unmarshal response body" could be modified to "couldn't process JSON response from server"
    1. "manifest not found" could be modified to "manifest not found. Please provide the path to the app manifest."

#### Implementation
The commands in scope of this issue are:

- [ ] `mesheryctl perf`
- [ ] `mesheryctl perf apply`
- [ ] `mesheryctl perf profile`
- [ ] `mesheryctl perf result`

#### Acceptance Tests
1. Contributor adhered to the [Meshery CLI Style Guide](https://docs.meshery.io/project/contributing/contributing-cli#meshery-cli-style-guide).

 --- 
  #### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)
  📚 For the **Reference** head over to this [pr](https://github.com/meshery/meshery/pull/5661).

### PR DESCRIPTION
**Pull request by gopivaibhav**
Signed-off-by: Gopi Vaibhav <gopivaibhav0@gmail.com>

**Notes for Reviewers**

The error message:
- Before
Screenshot from 2022-12-24 22-31-00
- After
Screenshot from 2022-12-25 00-07-53

This PR fixes #6269 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 76926bd588c7b15347e222b4f4bf2516961c15eb at: https://63a7489f68ae8925c4a36749--meshery-docs-preview.netlify.app

**Comment by gopivaibhav** [comment]
I am new to contribute in mesheryctl, which is using Cobra. I have searched a lot for creating custom error messages if the user doesn't provide the input for a field but couldn't find any. I would be glad if someone let me know how I can work on that.
Thank you!!

**Comment by leecalcote** [comment]
@rakshitgondwal,  @ujju20 it'd be good of you to offer review here.

**Comment by rakshitgondwal** [comment]
> I am new to contribute in mesheryctl, which is using Cobra. I have searched a lot for creating custom error messages if the user doesn't provide the input for a field but couldn't find any. I would be glad if someone let me know how I can work on that. Thank you!!

Hey! @gopivaibhav. Can you please brief me a little bit more about what kind of custom messages are you talking about?

**Comment by Chadha93** [comment]
@gopivaibhav  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by leecalcote** [comment]
@alphaX86 and @hexxdump, I wonder if you have any feedback for @gopivaibhav.

**Comment by rakshitgondwal** [comment]
> > > I am new to contribute in mesheryctl, which is using Cobra. I have searched a lot for creating custom error messages if the user doesn't provide the input for a field but couldn't find any. I would be glad if someone let me know how I can work on that. Thank you!!
> > 
> > 
> > Hey! @gopivaibhav. Can you please brief me a little bit more about what kind of custom messages are you talking about?
> 
> When we use the command **mesheryctl perf --token** but without giving the token, we will get an error that the flag is required from the Cobra. I have tried to customise this error but I couldn't find the exact solution for which I am looking for. And also, can you please let me know if there is anything else which I wanna change for this issue ?

I tried to solve this too but I am getting an error too. See [#6794 comments](https://github.com/meshery/meshery/pull/6794#issuecomment-1366139621) for reference

File: mesheryctl/internal/cli/root/perf/perf.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ mesheryctl perf result -o yaml
 			return cmd.Help()
 		}
 		if ok := utils.IsValidSubcommand(availableSubcommands, args[0]); !ok {
-			return errors.New(utils.PerfError(fmt.Sprintf("'%s' is a invalid command.  Use 'mesheryctl perf --help' to display usage guide.'\n", args[0])))
+			return errors.New(utils.PerfError(fmt.Sprintf("You are misspelling the command as '%s'. Use 'mesheryctl perf --help' to display usage guide.'\n", args[0])))
```
Comment (alphaX86) [inline mesheryctl/internal/cli/root/perf/perf.go:72]: This UX seems interesting... Could we implement for other commands too if such line exists?

Comment (alphaX86) [reply mesheryctl/internal/cli/root/perf/perf.go:72]: But wait... it seems it's not applicable to all use cases... we might need to reconsider this code line

Comment (gopivaibhav) [reply mesheryctl/internal/cli/root/perf/perf.go:72]: > But wait... it seems it's not applicable to all use cases coz even if user provides a "good" wrong subcommand it'll show as misspelt... we might need to reconsider this code line IMO

Yes, I have thought about this case but I have felt that they would see the user guide, which will be provided in the error message to get to know about the commands if they have used some invalid command. So,  what do you suggest I do?

**Comment by l5io** [comment]
🚀 Preview for commit 22a53df34c1370b03f49576f5bd0ea66d46288c7 at: https://63ecfa5685d1ce1ae6873e8b--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
@gopivaibhav what would you like to do about this PR?

**Comment by gopivaibhav** [comment]
> @gopivaibhav what would you like to do about this PR?

I am struck at a point. I wanna add this to the meeting minutes of the upcoming websites call if possible i.e. on Monday

**Comment by l5io** [comment]
🚀 Preview for commit d2beb15c303935cbc70999e0a5a08d3fd45136fd at: https://63ee11a174f71904a27db761--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
@gopivaibhav this Wednesday's Meshery Dev meeting would be great.

**Comment by alphaX86** [comment]
This PR has staled a lot due to inactivity. Closing this now...

**Pull request closed without merging**